src/FieldExchanger.jl

```diff
@@ -0,0 +1,210 @@
+
+    if turbulent_fluxes == FluxCalculator.CombinedAtmosGrid()
+        Interfacer.update_field!(atmos_sim, Val(:roughness_momentum), csf.z0m_S)
+        Interfacer.update_field!(atmos_sim, Val(:roughness_buoyancy), csf.z0b_S)
+        Interfacer.update_field!(atmos_sim, Val(:beta), csf.beta) # not in this version of atmos
+    end
+
+end
+
+"""
+    update_sim!(sim::SurfaceModelSimulation, csf, area_fraction = nothing)
+
+Updates the surface component model cache with the current coupler fields of F_A, F_R, F_E, P_liq, and ρ_sfc.
+
+# Arguments
+- `sim`: [Interfacer.SurfaceModelSimulation] containing a surface model simulation object.
+- `csf`: [NamedTuple] containing coupler fields.
+"""
+function update_sim!(sim::Interfacer.SurfaceModelSimulation, csf, area_fraction = nothing)
+
+    Interfacer.update_field!(sim, Val(:air_density), csf.ρ_sfc)
+
+    Interfacer.update_field!(
+        sim,
+        Val(:turbulent_energy_flux),
+        Regridder.binary_mask.(area_fraction, threshold = eps()) .* csf.F_A,
+    )
+
+    Interfacer.update_field!(
+        sim,
+        Val(:turbulent_moisture_flux),
+        Regridder.binary_mask.(area_fraction, threshold = eps()) .* csf.F_E,
+    )
+
+    Interfacer.update_field!(
+        sim,
+        Val(:radiative_energy_flux),
+        Regridder.binary_mask.(area_fraction, threshold = eps()) .* csf.F_R,
+    )
+
+    Interfacer.update_field!(sim, Val(:liquid_precipitation), .-csf.P_liq)
+    Interfacer.update_field!(sim, Val(:snow_precipitation), .-csf.P_snow)
+end
+
+"""
+    update_sim!(::SurfaceStub, csf, area_fraction)
+
+The stub surface simulation is not updated by this function.
+"""
+update_sim!(::Interfacer.SurfaceStub, csf, area_fraction) = nothing
+
+"""
+    update_model_sims!(model_sims, csf, turbulent_fluxes)
+
+Iterates `update_sim!` over all component model simulations saved in `cs.model_sims`.
+
+# Arguments
+- `model_sims`: [NamedTuple] containing `ComponentModelSimulation`s.
+- `csf`: [NamedTuple] containing coupler fields.
+- `turbulent_fluxes`: [TurbulentFluxPartition] denotes a flag for turbulent flux calculation.
+
+"""
+function update_model_sims!(model_sims, csf, turbulent_fluxes)
+    for sim in model_sims
+        if sim isa Interfacer.SurfaceModelSimulation
+            update_sim!(sim, csf, Interfacer.get_field(sim, Val(:area_fraction)))
+        else
+            update_sim!(sim, csf, turbulent_fluxes)
+        end
+    end
+end
+
+"""
+    reinit_model_sims!(model_sims)
+
+Iterates `reinit!` over all component model simulations saved in `cs.model_sims`.
+
+# Arguments
+- `model_sims`: [NamedTuple] containing `ComponentModelSimulation`s.
+
+"""
+function reinit_model_sims!(model_sims)
+    for sim in model_sims
+        reinit!(sim)
+    end
+end
+
+"""
+    reinit!(cs::SurfaceStub)
+
+The stub surface simulation is not updated by this function. Extends `SciMLBase.reinit!`.
+"""
+reinit!(::Interfacer.SurfaceStub) = nothing
```

Comment:
Do we need to define `reinit!` and `step!` for the other model types, or do we use methods inherited from SciMLBase?

Reply:
We define them in the component model-specific init files (e.g., `init_climaatmos`). Actually, this should be added to the docs!